### PR TITLE
Add topic permissions endpoints

### PR DESCRIPTION
**Pull request by acogoluegnes**
References rabbitmq/rabbitmq-server#505

### CODE REVIEW

File: src/rabbit_mgmt_dispatcher.erl [hunk 1]
```diff
@@ -110,16 +110,20 @@ dispatcher() ->
      {"/vhosts",                                               rabbit_mgmt_wm_vhosts, []},
      {"/vhosts/:vhost",                                        rabbit_mgmt_wm_vhost, []},
      {"/vhosts/:vhost/permissions",                            rabbit_mgmt_wm_permissions_vhost, []},
+     {"/vhosts/:vhost/topic-permissions",                      rabbit_mgmt_wm_topic_permissions_vhost, []},
```
Comment (michaelklishin) [inline src/rabbit_mgmt_dispatcher.erl]: Please use underscores, as we already do, e.g. `operator_policies`.

File: src/rabbit_mgmt_dispatcher.erl [hunk 2]
```diff
@@ -110,16 +110,20 @@ dispatcher() ->
...
      %% /connections/:connection is already taken, we cannot use our standard scheme here
      {"/vhosts/:vhost/connections",                            rabbit_mgmt_wm_connections_vhost, []},
      %% /channels/:channel is already taken, we cannot use our standard scheme here
      {"/vhosts/:vhost/channels",                               rabbit_mgmt_wm_channels_vhost, []},
      {"/users",                                                rabbit_mgmt_wm_users, []},
      {"/users/:user",                                          rabbit_mgmt_wm_user, []},
      {"/users/:user/permissions",                              rabbit_mgmt_wm_permissions_user, []},
+     {"/users/:user/topic-permissions",                        rabbit_mgmt_wm_topic_permissions_user, []},
```
Comment (michaelklishin) [inline src/rabbit_mgmt_dispatcher.erl]: Same as above.

File: src/rabbit_mgmt_wm_definitions.erl [hunk 3]
```diff
@@ -257,6 +259,7 @@ rw_state() ->
     [{users,              [name, password_hash, hashing_algorithm, tags]},
      {vhosts,             [name]},
      {permissions,        [user, vhost, configure, write, read]},
+     {topic_permissions,  [user, vhost, name, pattern]},
```
Comment (michaelklishin) [inline src/rabbit_mgmt_wm_definitions.erl]: I think `exchange` would be clearer than `name`.

File: src/rabbit_mgmt_dispatcher.erl (hunk 1 above)
Comment (acogoluegnes) [inline src/rabbit_mgmt_dispatcher.erl]: We also use dashes (e.g. `aliveness-test`, `cluster-name`, and also deprecated `all-configuration`), so I used them for `topic-permissions` and also recently for `global-parameters`. I don't mind changing those 2, but we won't be consistent anyway.
`operator_policies` and `vhost_limits` are in 3.7.0, so maybe we can still change them?

Comment (michaelklishin) [inline src/rabbit_mgmt_dispatcher.erl]: OK, we will change the new endpoints in master before it's too late then.

**Pull request merged**